examples/peft_llm_gnn/llm_gnn_model.py

```diff
@@ -0,0 +1,114 @@
+from graphstorm import model as gsmodel
+import torch
+import torch.nn as nn
+import os
+import torch.nn.functional as F
+from transformers import (
+    AutoConfig,
+    AutoModelForSequenceClassification,
+)
+from peft import LoraConfig, get_peft_model
+
+def get_lm_node_feats(g, lm_feat_names, lm_ntypes):
+    """ Collect language model related node features
+
+    Parameters
+    ----------
+    g: graph
+        Graph data.
+    lm_feat_names: list of str
+        A list of llm features
+    lm_ntypes: list of str
+        A list of node types.
+
+    Return
+    ------
+    A dict of dict of distributed tensor.
+        {Node type: {Feature name: Feature stored as distributed tensor}}
+    """
+    lm_feats = {}
+    for ntype in lm_ntypes:
+        lm_feats[ntype] = {}
+        for lm_fname in lm_feat_names:
+            if lm_fname in g.nodes[ntype].data:
+                lm_feats[ntype][lm_fname] = g.nodes[ntype].data[lm_fname]
+
+    return lm_feats
+
+
+class LLMGraphModel(gsmodel.GSgnnNodeModelBase):
+    """ LLMGNN Model Class
+
+    Parameters
+    ----------
+    """
+    def __init__(self, g, node_lm_configs, h_dim, out_dim, num_layers,
+                 target_ntype,     # the node type to be predict
+                 use_norm = True,   # use normalization or not, default is True
+                 alpha_l2norm = 0,
+                 lr = 0.001):
+        super(LLMGraphModel, self).__init__()
+        self.num_layers = num_layers
+        self.target_ntype=target_ntype
+        self.alpha_l2norm = alpha_l2norm
+        self.lr = lr
+        model_id = "facebook/opt-2.7b"
+        self.config = AutoConfig.from_pretrained(model_id, num_labels=out_dim)
+        base_model = AutoModelForSequenceClassification.from_pretrained(
+            model_id,config=self.config
+        )
+        peft_config = LoraConfig(task_type="SEQ_CLS", inference_mode=False, r=8, lora_alpha=16, lora_dropout=0.1)
```

Comment:
we probably should also make lora hyperparameters configurable.

Reply:
I plan to add configurable lora next PR with GNN encoded tokens. What do you think?